docs/sdk/src/reference_docs/frame_offchain_workers.rs

```diff
@@ -0,0 +1,115 @@
+//! state. This is why an HTTP ([`sp_runtime::offchain::http`]) API is readily provided to the
+//! offchain worker APIs. Because there is no need for determinism in this context.
+//!
+//! > A common mistake here is for novice developers to see this HTTP API, and imagine that
+//! > `polkadot-sdk` somehow magically solved the determinism in blockchains, and now a blockchain
+//! > can make HTTP calls and it will all work. This is absolutely NOT the case. An HTTP call made
+//! > by the offchain worker is non-deterministic by design. Blockchains can't and always won't be
+//! > able to perform non-deterministic operations such as making HTTP calls to a foreign server.
+//!
+//! ## FRAME's API
+//!
+//! [`frame`] provides a simple API through which pallets can define offchain worker functions. This
+//! is part of [`frame::traits::Hooks`], which is implemented as a part of
+//! [`frame::pallet_macros::hooks`].
+//!
+//! ```
+//! use frame::prelude::*;
+//!
+//! #[frame::pallet]
+//! pub mod pallet {
+//! 	use super::*;
+//!
+//! 	#[pallet::config]
+//! 	pub trait Config: frame_system::Config {}
+//!
+//! 	#[pallet::pallet]
+//! 	pub struct Pallet<T>(_);
+//!
+//! 	#[pallet::hooks]
+//! 	impl<T: Config> Hooks for Pallet<T> {
+//! 		fn offchain_worker(block_number: T::BlockNumber) {
+//! 			// ...
+//! 		}
+//! 	}
+//! }
+//! ```
+//!
+//! Additionally, [`sp_runtime::offchain`] provides a set of utilities that can be used to moderate
+//! the execution of offchain workers.
+//!
+//! ## Think Twice: Why Use Substrate's Offchain Workers?
+//!
+//! Consider the fact that in principle, an offchain worker code written using the above API is no
+//! different than an equivalent written with an _actual offchain interaction library_, such as
+//! [Polkadot-JS](https://polkadot.js.org/docs/), or any of the other ones listed [here](https://github.com/substrate-developer-hub/awesome-substrate?tab=readme-ov-file#client-libraries).
+//!
+//! They can both read from the state, and have no means of updating the state, other than the route
+//! of submitting an extrinsic to the chain. Therefore, it is worth thinking twice before embedding
+//! a logic as a part of Substrate's offchain worker API. Does it have to be there? can it not be a
+//! simple, actual offchain application that lives outside of the chain's WASM blob?
+//!
+//! Some of the reasons why you might want to do the opposite, and actually embed an offchain worker
+//! API into the WASM blob are:
+//!
+//! * Accessing the state is easier within the `offchain_worker` function, as it is already a part
+//!   of the runtime, and [`frame::pallet_macros::storage`] provides all the tools needed to read
+//!   the state. Other client libraries might provide varying degrees of capability here.
+//! * It will be updated in synchrony with the runtime. A Substrate's offchain application is part
+//!   of the same WASM blob, and is therefore guaranteed to be up to date.
+//!
```

Comment:
One of the notable properties, as you mentioned, is that OCW is part of the runtime API, making it more accessible. If your chain is operational, it implies that your OCW can be triggered by users. There should be use cases for this.

Reply:
Interestingly, this is exactly what the user in https://substrate.stackexchange.com/questions/11058/how-can-i-create-ocw-that-wont-activates-every-block-but-will-activates-only-w/11060#11060 asked for, but there is no easy way to do this yet, right? I suppose the best you can do is to link it to `state_call`, and trigger it.